### PR TITLE
SideNavigation: implement guidelines

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

SideNavigation: implement guidelines

### CODE REVIEW

File: packages/gestalt/src/SideNavigationTopItem.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ type Props = {|
 |};
 
 /**
- * Use [SideNavigation.TopItem](https://gestalt.pinterest.systems/sidenavigation#SideNavigation.TopItem) for navigation, when the SideNavigation.TopItem navigates to a new page or sectoin.
+ * Use [SideNavigation.TopItem](https://gestalt.pinterest.systems/sidenavigation#SideNavigation.TopItem) to redirect the user to a different page or section. SideNavigation.TopItem must be use at the top level of SideNavigation. It supports badges, icons, counters, and notifications.
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigationTopItem.js]: DONE

File: docs/components/PropTable.js [hunk 2]
```diff
@@ -59,16 +59,21 @@ function Td({
   shrink?: boolean,
   color?: 'default' | 'subtle',
 |}) {
+  let tdStyle = {
```
Comment (AlbertCarreras) [inline docs/components/PropTable.js:62]: unrelated to SideNav

File: docs/pages/sidenavigation.js [hunk 3]
```diff
@@ -69,49 +56,342 @@ export default function SideNavigationPage({
...
+`}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
       <AccessibilitySection name={generatedDocGen.SideNavigation?.displayName}>
         <MainSection.Subsection
           title="Active item"
-          description='SideNavigation.TopItem has an "active" state that visually identifies it. To set SideNavigation.TopItem to active pass "page" (page redirect) or "section" (selected content section). Use routing hooks from React.Router or other frameworks to identify the current route. For example, if the current pathname matches the SideNavigation.TopItem href, set SideNavigation.TopItem to "page".'
+          description={`SideNavigation.TopItem has an "active" state that visually identifies it. To set SideNavigation.TopItem to "active" set 'active="page"' (page redirect) or 'active="section"'. Use routing hooks from React.Router or other frameworks to identify the current route. For example, if the current pathname matches the SideNavigation.TopItem href, set SideNavigation.TopItem to "page". use example below as a reference.`}
```
Comment (ayeshakmaz) [inline docs/pages/sidenavigation.js:354]: typo at the end: Capitalize Use and add the: "Use the example..."

Comment (AlbertCarreras) [reply docs/pages/sidenavigation.js:354]: DONE

File: docs/pages/sidenavigation.js [hunk 4]
```diff
@@ -337,394 +644,453 @@ function Example() {
...
-      <SideNavigation.TopItem
-        href="https://gestalt.pinterest.systems/radiogroup"
-        label="RadioGroup"
-      />
-    </SideNavigation.Section>
-  </SideNavigation>
-</Box>
+<Flex width="100%">
+  <Box height={300} width={280} overflow="scroll">
+    <SideNavigation accessibilityLabel="Border example" showBorder>
```
[the remark this answers is not in the document]
Comment (ayeshakmaz) [reply docs/pages/sidenavigation.js:783]: ahh I think the width on the Box is too small, it's cutting it off

[the remark this answers is not in the document]
Comment (ayeshakmaz) [reply docs/pages/sidenavigation.js:783]: could just leave off the width entirely actually

[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/pages/sidenavigation.js:783]: Needed some rework the border...

[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/pages/sidenavigation.js:783]: DONE

File: packages/gestalt/src/SideNavigation.js [hunk 5]
```diff
@@ -56,35 +57,33 @@ export default function SideNavigation({
...
-            {footer ? (
-              <Flex direction="column" gap={4}>
-                <Divider />
-                <Box paddingX={4}>{footer}</Box>
-              </Flex>
-            ) : null}
-          </Flex>
-        </Box>
-      </div>
+      <Box minWidth={280} width={280}>
```
Comment (AlbertCarreras) [inline packages/gestalt/src/SideNavigation.js:60]: moving out width  to keep border correctly placed within 280 px width

File: docs/components/AppLayout.js [hunk 6]
```diff
@@ -25,9 +25,6 @@ export default function AppLayout({ children, colorScheme, isHomePage }: Props):
         <Box minWidth={MIN_NAV_WIDTH_PX}>
           <Navigation />
         </Box>
-
-        <Divider />
```
Comment (ayeshakmaz) [inline docs/components/AppLayout.js:29]: good catch!!

**Review by ayeshakmaz — APPROVED** [review]

File: docs/components/PropTable.js (hunk 2 above)
Comment (christianvuerings) [reply docs/components/PropTable.js:62]: @AlbertCarreras instead, could we avoid mutating an object (`tdStyle`) and do this logic inline? See fix at https://github.com/pinterest/gestalt/pull/2220/files#diff-0520d70de71d8ac5e7747ab488703c88263481a4b3ce3579d3597cef8a940feaL62

The previous hydration issue was caused by the fact that we didn't always return `null`

```
orderBottom: border && '1px solid #ddd',
```
should have been
```
orderBottom: border ? '1px solid #ddd' : null,
```

and
```
width: shrink ? '1px' : '',
```
should have been
```
width: shrink ? '1px' : null,
```

**Pull request merged**